lvmd/lvservice.go

```diff
@@ -41,7 +43,27 @@ func (s *lvService) CreateLV(_ context.Context, req *proto.CreateLVRequest) (*pr
 		return nil, err
 	}
 	requested := req.GetSizeGb() << 30
-	free, err := vg.Free()
+	free := uint64(0)
+	var pool *command.ThinPool
+	switch dc.Type {
+	case TypeThick:
+		free, err = vg.Free()
+	case TypeThin:
+		pool, err = vg.FindPool(dc.ThinPoolConfig.Name)
+		if err != nil {
+			return nil, err
+		}
+		tpu, err := pool.Free()
+		if err != nil {
+			return nil, err
+		}
+		free = uint64(math.Ceil(dc.ThinPoolConfig.OverprovisionRatio*float64(tpu.SizeBytes))) - tpu.VirtualBytes
```

Comment:
```suggestion
		free = uint64(math.Ceil(dc.ThinPoolConfig.OverprovisionRatio*float64(tpu.FreeBytes)))
```
`OverprovisionRatio * SizeBytes - VirtualBytes` is not the virtual free size. `FreeBytes` is near the actual free size, it is suitable to calculate free size.

Reply:
- I need to disagree with this statement
- SizeBytes is a constant unless thinpool is extended and we should be calculating the overprovisioned size based on actual thinpool size but not on the moving target FreeBytes which depends on data & meta percent used
- our intention is over-provisioned size should reduce when there are multiple thin lvs created in the thinpool
- however, if we use freebytes in the calculation, over-provisioned size will only be reduced based on actual usage of those thin lvs

Reply:
@leelavg 
I understood the intent of the formula, but I have a question. Is this formula works normally when snapshots exists?

Reply:
- As a gist, overprovision size is being used for only pod scheduling and better we take snapshot size as well into consideration
- there's a possibility that some snapshots will be activated (~clones) for read/write and they'll behave as a normal thin lv at some point
- having all said, data and metadata percent on thinpool is the key for monitoring and we are exporting those metrics
- so I propose we don't change the current formula and revisit if there need to be some changes after snapshots is merged, sounds good?

Reply:
I think `OverprovisionRatio * SizeBytes - VirtualBytes ` - `data size consumed by the snapshot` is better. About snapshot size, could't you get the value by following option?

```
$ lvs -o help
  Logical Volume Fields
  ---------------------
...
    origin_size            - For snapshots, the size of the origin device of this LV.
...
```

> I want to handle this after looking at snapshot implementation

I agree with you.

Reply:
- yes, that's the idea after snapshot gets it, cause existing one doesn't effect the operations in any way
- will definitely get back after snapshot PR